### PR TITLE
[dockers]: Upgrade LLDP docker to stretch build

### PR DESCRIPTION
**Pull request by samaity**
Signed-off-by: Sangita Maity [samaity@linkedin.com](mailto:samaity@linkedin.com)

**- What I did**
Move LLDP docker to stretch based. **Requires below mentioned pull request** to build libsnmp debian packages as the LLDP docker depends on libsnmp debian package.

1. [https://github.com/Azure/sonic-buildimage/pull/2620](https://github.com/Azure/sonic-buildimage/pull/2620)

> This is the pull request for  SNMP docker upgrade to stretch build.

**- How I did it**
```
samaity@Server09:~/github_srcCode/lldp_srcCode_28feb/sonic-buildimage$ BLDENV=stretch make stretch
+++ Making stretch +++
BLDENV=stretch make -f Makefile.work stretch
make[1]: Entering directory '/home/samaity/github_srcCode/lldp_srcCode_28feb/sonic-buildimage'
SONiC Build System

Build Configuration
"CONFIGURED_PLATFORM"             : "broadcom"
"SONIC_CONFIG_PRINT_DEPENDENCIES" : ""
"SONIC_BUILD_JOBS"                : "1"
"SONIC_CONFIG_MAKE_JOBS"          : "24"
"USERNAME"                        : "admin"
"PASSWORD"                        : "YourPaSsWoRd"
"ENABLE_DHCP_GRAPH_SERVICE"       : ""
"SHUTDOWN_BGP_ON_START"           : ""
"ENABLE_PFCWD_ON_START"           : ""
"INSTALL_DEBUG_TOOLS"             : ""
"ROUTING_STACK"                   : "quagga"
"ENABLE_SYNCD_RPC"                : ""
"ENABLE_ORGANIZATION_EXTENSIONS"  : "y"
"HTTP_PROXY"                      : ""
"HTTPS_PROXY"                     : ""
"ENABLE_SYSTEM_TELEMETRY"         : ""
"SONIC_DEBUGGING_ON"              : ""
"SONIC_PROFILING_ON"              : ""
"KERNEL_PROCURE_METHOD"           : "build"
"BUILD_TIMESTAMP"                 : "20190301.121613"
"BLDENV"                          : "stretch"

[ 01 ] [ target/docker-lldp-sv2.gz ]

```
**- How to verify it**
### Created images base on newly created docker-lldp-sv2.gz file.
```
admin@lnos-x1-a-csw04:~$ docker images
REPOSITORY                 TAG                 IMAGE ID            CREATED             SIZE
docker-lldp-sv2            latest              f02adda41938        8 minutes ago       310.3 MB
```
### Checked all the service under LLDP docker.
```
admin@lnos-x1-a-csw04:~$ docker exec -it lldp bash
root@lnos-x1-a-csw04:/# ps -ef
UID        PID  PPID  C STIME TTY          TIME CMD
root         1     0  0 20:31 ?        00:00:00 /usr/bin/python /usr/bin/supervisord
root        17     1  0 20:31 ?        00:00:00 /usr/sbin/rsyslogd -n
root        22     1  0 20:31 ?        00:00:00 lldpd: monitor.
_lldpd      24    22  0 20:31 ?        00:00:00 lldpd: 2 neighbors.
root        26     1  0 20:31 ?        00:00:00 python2 -m lldp_syncd
root        32     1  0 20:32 ?        00:00:00 python /usr/bin/lldpmgrd
root        45     0  0 20:32 ?        00:00:00 bash
root        51    45  0 20:32 ?        00:00:00 ps -ef
```
### Checked Supervisor status inside the LLDP docker. All services are running fine.
```
root@lnos-x1-a-csw04:/# supervisorctl  status
lldp-syncd                       RUNNING   pid 26, uptime 0:00:50
lldpd                            RUNNING   pid 22, uptime 0:00:51
lldpmgrd                         RUNNING   pid 32, uptime 0:00:49
rsyslogd                         RUNNING   pid 17, uptime 0:00:52
start.sh                         EXITED    Mar 01 08:32 PM

```
### All the debian packages of the stretch version required for LLDP on a BUILD server->
```
samaity@Server09:~/github_srcCode/lldp_srcCode_28feb/sonic-buildimage/target/debs/stretch$ ls -lrt *libsnmp*
-rw-r--r-- 1 samaity samaity  223342 Feb 28 14:36 libsnmp-base_5.7.3+dfsg-1.5_all.deb.log
-rw-r--r-- 1 samaity samaity 1594854 Feb 28 16:51 libsnmp-base_5.7.3+dfsg-1.7+deb9u1_all.deb
-rw-r--r-- 1 samaity samaity 2267844 Feb 28 16:51 libsnmp30_5.7.3+dfsg-1.7+deb9u1_amd64.deb
-rw-r--r-- 1 samaity samaity 2252048 Feb 28 16:51 libsnmp30-dbg_5.7.3+dfsg-1.7+deb9u1_amd64.deb
-rw-r--r-- 1 samaity samaity 1054846 Feb 28 16:52 libsnmp-dev_5.7.3+dfsg-1.7+deb9u1_amd64.deb
-rw-r--r-- 1 samaity samaity 1497778 Feb 28 16:52 libsnmp-perl_5.7.3+dfsg-1.7+deb9u1_amd64.deb
-rw-r--r-- 1 samaity samaity 1092639 Feb 28 16:52 libsnmp-base_5.7.3+dfsg-1.7+deb9u1_all.deb.log
-rw-r--r-- 1 samaity samaity     322 Mar  1 11:54 libsnmp-dev_5.7.3+dfsg-1.7+deb9u1_amd64.deb-install.log
-rw-r--r-- 1 samaity samaity     323 Mar  1 12:25 libsnmp-base_5.7.3+dfsg-1.7+deb9u1_all.deb-install.log
-rw-r--r-- 1 samaity samaity     335 Mar  1 12:25 libsnmp30_5.7.3+dfsg-1.7+deb9u1_amd64.deb-install.log
samaity@Server09:~/github_srcCode/lldp_srcCode_28feb/sonic-buildimage/target/debs/stretch$ ls -lrt *lldp*
-rw-r--r-- 1 samaity samaity  19802 Mar  1 11:55 liblldpctl-dev_0.9.6-1_amd64.deb
-rw-r--r-- 1 samaity samaity 150408 Mar  1 11:55 lldpd_0.9.6-1_amd64.deb
-rw-r--r-- 1 samaity samaity 246928 Mar  1 11:55 lldpd_0.9.6-1_amd64.deb.log

```
**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: dockers/docker-lldp-sv2/Dockerfile.j2 [hunk 1]
```diff
@@ -10,7 +10,7 @@ ENV DEBIAN_FRONTEND=noninteractive
 RUN apt-get update
 
 # Install dependencies
-RUN apt-get install -y python-pip libbsd0 libevent-2.0-5 libjansson4 libwrap0 libxml2 libpci3 libperl5.20 libpython2.7
+RUN apt-get install -y adduser init-system-helpers libbsd0 libevent-2.0-5 libjansson4 libsensors4 libwrap0 libxml2 libpci3 python-pip lsb-base libpython2.7
```
Comment (jleveque) [inline dockers/docker-lldp-sv2/Dockerfile.j2]: Can you please explain why we now need to install adduser, init-system-helpers, libsensors4 and lsb-base?

Comment (samaity) [reply dockers/docker-lldp-sv2/Dockerfile.j2]: Hey Joe,
I just followed the requirement to build lldpd as per below docs.
**Link:** [https://packages.debian.org/stretch/lldpd](url)

so, added those which were not there in the dockerfile.

Comment (jleveque) [reply dockers/docker-lldp-sv2/Dockerfile.j2]: Thank you for paying attention to these dependencies. However, I'm not sure if it's necessary to install them explicitly here. Did you try building an image without adding them? The `RUN dpkg_apt() ...` lines below should install all built packages _as well as_ their dependencies. I checked the lldpd dependencies for Debian Jessie and most of the same dependencies are listed, however we never explicitly installed them, yet everything built and worked just fine.

The idea behind adding the `RUN dpkg_apt() ...` lines is that we will not need to maintain the list of dependencies going forward. Actually, I wonder if anything is necessary in that list besides `python-pip` simply for installing Python Wheels. Could you please see how minimal you can make that list?

Comment (samaity) [reply dockers/docker-lldp-sv2/Dockerfile.j2]: Hey Joe, 
I just removed all the extra packages I installed for lldpd in the DockerFile. You are correct . We don't need this explicitly.  I will remove this. Thanks for your explanation!

Comment (jleveque) [reply dockers/docker-lldp-sv2/Dockerfile.j2]: You're welcome. Are you able to remove any of the pre-existing dependent packages, also?

**Comment by zhenggen-xu** [comment]
@samaity As discussed, let's keep the source code as is and give it a try first.

**Comment by zhenggen-xu** [comment]
retest this please

**Comment by samaity** [comment]
retest this please

**Comment by lguohan** [comment]
retest this please

**Pull request merged**